{
  "cells": [
    {
      "cell_type": "code",
      "execution_count": 2,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "J9eWRmSHprDv",
        "outputId": "24dd7666-f23f-4791-c939-9e0cf8dd2805"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Cloning into 'Vietnamese-sentiment-analysis'...\n",
            "remote: Enumerating objects: 614, done.\u001b[K\n",
            "remote: Counting objects: 100% (178/178), done.\u001b[K\n",
            "remote: Compressing objects: 100% (117/117), done.\u001b[K\n",
            "remote: Total 614 (delta 118), reused 111 (delta 61), pack-reused 436\u001b[K\n",
            "Receiving objects: 100% (614/614), 2.03 MiB | 5.23 MiB/s, done.\n",
            "Resolving deltas: 100% (403/403), done.\n"
          ]
        }
      ],
      "source": [
        "!git clone https://github.com/minhquan6203/DS201"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "mX3AhUjfC0Gd"
      },
      "source": [
        "# prepare data"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 3,
      "metadata": {
        "id": "z5GTTO6WcGOA"
      },
      "outputs": [],
      "source": [
        "import pandas as pd\n",
        "import os\n",
        "train=pd.read_csv('/content/DS201/TH5/data/UIT-VSFC/train.csv')\n",
        "dev=pd.read_csv('/content/DS201/TH5/data/UIT-VSFC/dev.csv')\n",
        "test=pd.read_csv('/content/DS201/TH5/data/UIT-VSFC/test.csv')\n",
        "os.makedirs('data',exist_ok=True)\n",
        "train.to_csv('data/train.csv',index=False)\n",
        "dev.to_csv('data/dev.csv',index=False)\n",
        "test.to_csv('data/test.csv',index=False)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "rZ-LfApSzcxT"
      },
      "outputs": [],
      "source": [
        "# train=train.rename(columns={'Emotion':'sentiment','Sentence':'sentence'})\n",
        "# dev=dev.rename(columns={'Emotion':'sentiment','Sentence':'sentence'})\n",
        "# test=test.rename(columns={'Emotion':'sentiment','Sentence':'sentence'})"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 4,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "1OfUZzMJqFNS",
        "outputId": "c338bfb2-f7e8-4d39-95fb-0936bc029cf0"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Collecting yacs\n",
            "  Downloading yacs-0.1.8-py3-none-any.whl (14 kB)\n",
            "Requirement already satisfied: PyYAML in /usr/local/lib/python3.10/dist-packages (from yacs) (6.0.1)\n",
            "Installing collected packages: yacs\n",
            "Successfully installed yacs-0.1.8\n",
            "Collecting transformers==4.33.0\n",
            "  Downloading transformers-4.33.0-py3-none-any.whl (7.6 MB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m7.6/7.6 MB\u001b[0m \u001b[31m23.1 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hRequirement already satisfied: filelock in /usr/local/lib/python3.10/dist-packages (from transformers==4.33.0) (3.13.1)\n",
            "Requirement already satisfied: huggingface-hub<1.0,>=0.15.1 in /usr/local/lib/python3.10/dist-packages (from transformers==4.33.0) (0.19.4)\n",
            "Requirement already satisfied: numpy>=1.17 in /usr/local/lib/python3.10/dist-packages (from transformers==4.33.0) (1.23.5)\n",
            "Requirement already satisfied: packaging>=20.0 in /usr/local/lib/python3.10/dist-packages (from transformers==4.33.0) (23.2)\n",
            "Requirement already satisfied: pyyaml>=5.1 in /usr/local/lib/python3.10/dist-packages (from transformers==4.33.0) (6.0.1)\n",
            "Requirement already satisfied: regex!=2019.12.17 in /usr/local/lib/python3.10/dist-packages (from transformers==4.33.0) (2023.6.3)\n",
            "Requirement already satisfied: requests in /usr/local/lib/python3.10/dist-packages (from transformers==4.33.0) (2.31.0)\n",
            "Collecting tokenizers!=0.11.3,<0.14,>=0.11.1 (from transformers==4.33.0)\n",
            "  Downloading tokenizers-0.13.3-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (7.8 MB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m7.8/7.8 MB\u001b[0m \u001b[31m61.9 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hRequirement already satisfied: safetensors>=0.3.1 in /usr/local/lib/python3.10/dist-packages (from transformers==4.33.0) (0.4.1)\n",
            "Requirement already satisfied: tqdm>=4.27 in /usr/local/lib/python3.10/dist-packages (from transformers==4.33.0) (4.66.1)\n",
            "Requirement already satisfied: fsspec>=2023.5.0 in /usr/local/lib/python3.10/dist-packages (from huggingface-hub<1.0,>=0.15.1->transformers==4.33.0) (2023.6.0)\n",
            "Requirement already satisfied: typing-extensions>=3.7.4.3 in /usr/local/lib/python3.10/dist-packages (from huggingface-hub<1.0,>=0.15.1->transformers==4.33.0) (4.5.0)\n",
            "Requirement already satisfied: charset-normalizer<4,>=2 in /usr/local/lib/python3.10/dist-packages (from requests->transformers==4.33.0) (3.3.2)\n",
            "Requirement already satisfied: idna<4,>=2.5 in /usr/local/lib/python3.10/dist-packages (from requests->transformers==4.33.0) (3.6)\n",
            "Requirement already satisfied: urllib3<3,>=1.21.1 in /usr/local/lib/python3.10/dist-packages (from requests->transformers==4.33.0) (2.0.7)\n",
            "Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.10/dist-packages (from requests->transformers==4.33.0) (2023.11.17)\n",
            "Installing collected packages: tokenizers, transformers\n",
            "  Attempting uninstall: tokenizers\n",
            "    Found existing installation: tokenizers 0.15.0\n",
            "    Uninstalling tokenizers-0.15.0:\n",
            "      Successfully uninstalled tokenizers-0.15.0\n",
            "  Attempting uninstall: transformers\n",
            "    Found existing installation: transformers 4.35.2\n",
            "    Uninstalling transformers-4.35.2:\n",
            "      Successfully uninstalled transformers-4.35.2\n",
            "Successfully installed tokenizers-0.13.3 transformers-4.33.0\n",
            "Collecting SentencePiece\n",
            "  Downloading sentencepiece-0.1.99-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (1.3 MB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m1.3/1.3 MB\u001b[0m \u001b[31m8.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hInstalling collected packages: SentencePiece\n",
            "Successfully installed SentencePiece-0.1.99\n",
            "Requirement already satisfied: transformers in /usr/local/lib/python3.10/dist-packages (4.33.0)\n",
            "Collecting datasets\n",
            "  Downloading datasets-2.15.0-py3-none-any.whl (521 kB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m521.2/521.2 kB\u001b[0m \u001b[31m7.9 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hRequirement already satisfied: filelock in /usr/local/lib/python3.10/dist-packages (from transformers) (3.13.1)\n",
            "Requirement already satisfied: huggingface-hub<1.0,>=0.15.1 in /usr/local/lib/python3.10/dist-packages (from transformers) (0.19.4)\n",
            "Requirement already satisfied: numpy>=1.17 in /usr/local/lib/python3.10/dist-packages (from transformers) (1.23.5)\n",
            "Requirement already satisfied: packaging>=20.0 in /usr/local/lib/python3.10/dist-packages (from transformers) (23.2)\n",
            "Requirement already satisfied: pyyaml>=5.1 in /usr/local/lib/python3.10/dist-packages (from transformers) (6.0.1)\n",
            "Requirement already satisfied: regex!=2019.12.17 in /usr/local/lib/python3.10/dist-packages (from transformers) (2023.6.3)\n",
            "Requirement already satisfied: requests in /usr/local/lib/python3.10/dist-packages (from transformers) (2.31.0)\n",
            "Requirement already satisfied: tokenizers!=0.11.3,<0.14,>=0.11.1 in /usr/local/lib/python3.10/dist-packages (from transformers) (0.13.3)\n",
            "Requirement already satisfied: safetensors>=0.3.1 in /usr/local/lib/python3.10/dist-packages (from transformers) (0.4.1)\n",
            "Requirement already satisfied: tqdm>=4.27 in /usr/local/lib/python3.10/dist-packages (from transformers) (4.66.1)\n",
            "Requirement already satisfied: pyarrow>=8.0.0 in /usr/local/lib/python3.10/dist-packages (from datasets) (10.0.1)\n",
            "Collecting pyarrow-hotfix (from datasets)\n",
            "  Downloading pyarrow_hotfix-0.6-py3-none-any.whl (7.9 kB)\n",
            "Collecting dill<0.3.8,>=0.3.0 (from datasets)\n",
            "  Downloading dill-0.3.7-py3-none-any.whl (115 kB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m115.3/115.3 kB\u001b[0m \u001b[31m8.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hRequirement already satisfied: pandas in /usr/local/lib/python3.10/dist-packages (from datasets) (1.5.3)\n",
            "Requirement already satisfied: xxhash in /usr/local/lib/python3.10/dist-packages (from datasets) (3.4.1)\n",
            "Collecting multiprocess (from datasets)\n",
            "  Downloading multiprocess-0.70.15-py310-none-any.whl (134 kB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m134.8/134.8 kB\u001b[0m \u001b[31m9.8 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hRequirement already satisfied: fsspec[http]<=2023.10.0,>=2023.1.0 in /usr/local/lib/python3.10/dist-packages (from datasets) (2023.6.0)\n",
            "Requirement already satisfied: aiohttp in /usr/local/lib/python3.10/dist-packages (from datasets) (3.9.1)\n",
            "Requirement already satisfied: attrs>=17.3.0 in /usr/local/lib/python3.10/dist-packages (from aiohttp->datasets) (23.1.0)\n",
            "Requirement already satisfied: multidict<7.0,>=4.5 in /usr/local/lib/python3.10/dist-packages (from aiohttp->datasets) (6.0.4)\n",
            "Requirement already satisfied: yarl<2.0,>=1.0 in /usr/local/lib/python3.10/dist-packages (from aiohttp->datasets) (1.9.3)\n",
            "Requirement already satisfied: frozenlist>=1.1.1 in /usr/local/lib/python3.10/dist-packages (from aiohttp->datasets) (1.4.0)\n",
            "Requirement already satisfied: aiosignal>=1.1.2 in /usr/local/lib/python3.10/dist-packages (from aiohttp->datasets) (1.3.1)\n",
            "Requirement already satisfied: async-timeout<5.0,>=4.0 in /usr/local/lib/python3.10/dist-packages (from aiohttp->datasets) (4.0.3)\n",
            "Requirement already satisfied: typing-extensions>=3.7.4.3 in /usr/local/lib/python3.10/dist-packages (from huggingface-hub<1.0,>=0.15.1->transformers) (4.5.0)\n",
            "Requirement already satisfied: charset-normalizer<4,>=2 in /usr/local/lib/python3.10/dist-packages (from requests->transformers) (3.3.2)\n",
            "Requirement already satisfied: idna<4,>=2.5 in /usr/local/lib/python3.10/dist-packages (from requests->transformers) (3.6)\n",
            "Requirement already satisfied: urllib3<3,>=1.21.1 in /usr/local/lib/python3.10/dist-packages (from requests->transformers) (2.0.7)\n",
            "Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.10/dist-packages (from requests->transformers) (2023.11.17)\n",
            "Requirement already satisfied: python-dateutil>=2.8.1 in /usr/local/lib/python3.10/dist-packages (from pandas->datasets) (2.8.2)\n",
            "Requirement already satisfied: pytz>=2020.1 in /usr/local/lib/python3.10/dist-packages (from pandas->datasets) (2023.3.post1)\n",
            "Requirement already satisfied: six>=1.5 in /usr/local/lib/python3.10/dist-packages (from python-dateutil>=2.8.1->pandas->datasets) (1.16.0)\n",
            "Installing collected packages: pyarrow-hotfix, dill, multiprocess, datasets\n",
            "Successfully installed datasets-2.15.0 dill-0.3.7 multiprocess-0.70.15 pyarrow-hotfix-0.6\n",
            "Collecting accelerate\n",
            "  Downloading accelerate-0.25.0-py3-none-any.whl (265 kB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m265.7/265.7 kB\u001b[0m \u001b[31m6.1 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hRequirement already satisfied: numpy>=1.17 in /usr/local/lib/python3.10/dist-packages (from accelerate) (1.23.5)\n",
            "Requirement already satisfied: packaging>=20.0 in /usr/local/lib/python3.10/dist-packages (from accelerate) (23.2)\n",
            "Requirement already satisfied: psutil in /usr/local/lib/python3.10/dist-packages (from accelerate) (5.9.5)\n",
            "Requirement already satisfied: pyyaml in /usr/local/lib/python3.10/dist-packages (from accelerate) (6.0.1)\n",
            "Requirement already satisfied: torch>=1.10.0 in /usr/local/lib/python3.10/dist-packages (from accelerate) (2.1.0+cu118)\n",
            "Requirement already satisfied: huggingface-hub in /usr/local/lib/python3.10/dist-packages (from accelerate) (0.19.4)\n",
            "Requirement already satisfied: safetensors>=0.3.1 in /usr/local/lib/python3.10/dist-packages (from accelerate) (0.4.1)\n",
            "Requirement already satisfied: filelock in /usr/local/lib/python3.10/dist-packages (from torch>=1.10.0->accelerate) (3.13.1)\n",
            "Requirement already satisfied: typing-extensions in /usr/local/lib/python3.10/dist-packages (from torch>=1.10.0->accelerate) (4.5.0)\n",
            "Requirement already satisfied: sympy in /usr/local/lib/python3.10/dist-packages (from torch>=1.10.0->accelerate) (1.12)\n",
            "Requirement already satisfied: networkx in /usr/local/lib/python3.10/dist-packages (from torch>=1.10.0->accelerate) (3.2.1)\n",
            "Requirement already satisfied: jinja2 in /usr/local/lib/python3.10/dist-packages (from torch>=1.10.0->accelerate) (3.1.2)\n",
            "Requirement already satisfied: fsspec in /usr/local/lib/python3.10/dist-packages (from torch>=1.10.0->accelerate) (2023.6.0)\n",
            "Requirement already satisfied: triton==2.1.0 in /usr/local/lib/python3.10/dist-packages (from torch>=1.10.0->accelerate) (2.1.0)\n",
            "Requirement already satisfied: requests in /usr/local/lib/python3.10/dist-packages (from huggingface-hub->accelerate) (2.31.0)\n",
            "Requirement already satisfied: tqdm>=4.42.1 in /usr/local/lib/python3.10/dist-packages (from huggingface-hub->accelerate) (4.66.1)\n",
            "Requirement already satisfied: MarkupSafe>=2.0 in /usr/local/lib/python3.10/dist-packages (from jinja2->torch>=1.10.0->accelerate) (2.1.3)\n",
            "Requirement already satisfied: charset-normalizer<4,>=2 in /usr/local/lib/python3.10/dist-packages (from requests->huggingface-hub->accelerate) (3.3.2)\n",
            "Requirement already satisfied: idna<4,>=2.5 in /usr/local/lib/python3.10/dist-packages (from requests->huggingface-hub->accelerate) (3.6)\n",
            "Requirement already satisfied: urllib3<3,>=1.21.1 in /usr/local/lib/python3.10/dist-packages (from requests->huggingface-hub->accelerate) (2.0.7)\n",
            "Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.10/dist-packages (from requests->huggingface-hub->accelerate) (2023.11.17)\n",
            "Requirement already satisfied: mpmath>=0.19 in /usr/local/lib/python3.10/dist-packages (from sympy->torch>=1.10.0->accelerate) (1.3.0)\n",
            "Installing collected packages: accelerate\n",
            "Successfully installed accelerate-0.25.0\n"
          ]
        },
        {
          "name": "stderr",
          "output_type": "stream",
          "text": [
            "[nltk_data] Downloading package wordnet to /root/nltk_data...\n",
            "[nltk_data] Downloading package omw-1.4 to /root/nltk_data...\n"
          ]
        },
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Collecting peft\n",
            "  Downloading peft-0.7.0-py3-none-any.whl (168 kB)\n",
            "\u001b[?25l     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m0.0/168.3 kB\u001b[0m \u001b[31m?\u001b[0m eta \u001b[36m-:--:--\u001b[0m\r\u001b[2K     \u001b[91m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m\u001b[90m╺\u001b[0m\u001b[90m━━━━━\u001b[0m \u001b[32m143.4/168.3 kB\u001b[0m \u001b[31m4.2 MB/s\u001b[0m eta \u001b[36m0:00:01\u001b[0m\r\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m168.3/168.3 kB\u001b[0m \u001b[31m3.8 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hRequirement already satisfied: numpy>=1.17 in /usr/local/lib/python3.10/dist-packages (from peft) (1.23.5)\n",
            "Requirement already satisfied: packaging>=20.0 in /usr/local/lib/python3.10/dist-packages (from peft) (23.2)\n",
            "Requirement already satisfied: psutil in /usr/local/lib/python3.10/dist-packages (from peft) (5.9.5)\n",
            "Requirement already satisfied: pyyaml in /usr/local/lib/python3.10/dist-packages (from peft) (6.0.1)\n",
            "Requirement already satisfied: torch>=1.13.0 in /usr/local/lib/python3.10/dist-packages (from peft) (2.1.0+cu118)\n",
            "Requirement already satisfied: transformers in /usr/local/lib/python3.10/dist-packages (from peft) (4.33.0)\n",
            "Requirement already satisfied: tqdm in /usr/local/lib/python3.10/dist-packages (from peft) (4.66.1)\n",
            "Requirement already satisfied: accelerate>=0.21.0 in /usr/local/lib/python3.10/dist-packages (from peft) (0.25.0)\n",
            "Requirement already satisfied: safetensors in /usr/local/lib/python3.10/dist-packages (from peft) (0.4.1)\n",
            "Requirement already satisfied: huggingface-hub>=0.17.0 in /usr/local/lib/python3.10/dist-packages (from peft) (0.19.4)\n",
            "Requirement already satisfied: filelock in /usr/local/lib/python3.10/dist-packages (from huggingface-hub>=0.17.0->peft) (3.13.1)\n",
            "Requirement already satisfied: fsspec>=2023.5.0 in /usr/local/lib/python3.10/dist-packages (from huggingface-hub>=0.17.0->peft) (2023.6.0)\n",
            "Requirement already satisfied: requests in /usr/local/lib/python3.10/dist-packages (from huggingface-hub>=0.17.0->peft) (2.31.0)\n",
            "Requirement already satisfied: typing-extensions>=3.7.4.3 in /usr/local/lib/python3.10/dist-packages (from huggingface-hub>=0.17.0->peft) (4.5.0)\n",
            "Requirement already satisfied: sympy in /usr/local/lib/python3.10/dist-packages (from torch>=1.13.0->peft) (1.12)\n",
            "Requirement already satisfied: networkx in /usr/local/lib/python3.10/dist-packages (from torch>=1.13.0->peft) (3.2.1)\n",
            "Requirement already satisfied: jinja2 in /usr/local/lib/python3.10/dist-packages (from torch>=1.13.0->peft) (3.1.2)\n",
            "Requirement already satisfied: triton==2.1.0 in /usr/local/lib/python3.10/dist-packages (from torch>=1.13.0->peft) (2.1.0)\n",
            "Requirement already satisfied: regex!=2019.12.17 in /usr/local/lib/python3.10/dist-packages (from transformers->peft) (2023.6.3)\n",
            "Requirement already satisfied: tokenizers!=0.11.3,<0.14,>=0.11.1 in /usr/local/lib/python3.10/dist-packages (from transformers->peft) (0.13.3)\n",
            "Requirement already satisfied: MarkupSafe>=2.0 in /usr/local/lib/python3.10/dist-packages (from jinja2->torch>=1.13.0->peft) (2.1.3)\n",
            "Requirement already satisfied: charset-normalizer<4,>=2 in /usr/local/lib/python3.10/dist-packages (from requests->huggingface-hub>=0.17.0->peft) (3.3.2)\n",
            "Requirement already satisfied: idna<4,>=2.5 in /usr/local/lib/python3.10/dist-packages (from requests->huggingface-hub>=0.17.0->peft) (3.6)\n",
            "Requirement already satisfied: urllib3<3,>=1.21.1 in /usr/local/lib/python3.10/dist-packages (from requests->huggingface-hub>=0.17.0->peft) (2.0.7)\n",
            "Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.10/dist-packages (from requests->huggingface-hub>=0.17.0->peft) (2023.11.17)\n",
            "Requirement already satisfied: mpmath>=0.19 in /usr/local/lib/python3.10/dist-packages (from sympy->torch>=1.13.0->peft) (1.3.0)\n",
            "Installing collected packages: peft\n",
            "Successfully installed peft-0.7.0\n"
          ]
        }
      ],
      "source": [
        "!pip install yacs\n",
        "!pip install transformers==4.33.0\n",
        "!pip install SentencePiece\n",
        "!pip install transformers datasets\n",
        "!pip install --upgrade accelerate\n",
        "import nltk\n",
        "nltk.download('wordnet')\n",
        "nltk.download('omw-1.4')"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "bF-Wsk-oKqyG"
      },
      "source": [
        "# training"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 6,
      "metadata": {
        "id": "Jj00XaIPr24q"
      },
      "outputs": [],
      "source": [
        "import os\n",
        "os.environ[\"TOKENIZERS_PARALLELISM\"] = \"false\""
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "VrSrVwoaemiR"
      },
      "outputs": [],
      "source": [
        "!rm -rf /content/checkpoint"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 7,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "LROxbrV0r6Gn",
        "outputId": "580952cd-8f16-453c-89dd-6baeb76c28b4"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "/content\n",
            "2023-12-12 08:42:38.237802: E tensorflow/compiler/xla/stream_executor/cuda/cuda_dnn.cc:9342] Unable to register cuDNN factory: Attempting to register factory for plugin cuDNN when one has already been registered\n",
            "2023-12-12 08:42:38.237864: E tensorflow/compiler/xla/stream_executor/cuda/cuda_fft.cc:609] Unable to register cuFFT factory: Attempting to register factory for plugin cuFFT when one has already been registered\n",
            "2023-12-12 08:42:38.237907: E tensorflow/compiler/xla/stream_executor/cuda/cuda_blas.cc:1518] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered\n",
            "2023-12-12 08:42:40.080267: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Could not find TensorRT\n",
            "Downloading data files: 100% 3/3 [00:00<00:00, 16152.65it/s]\n",
            "Extracting data files: 100% 3/3 [00:00<00:00, 1645.90it/s]\n",
            "Generating train split: 11426 examples [00:00, 114628.51 examples/s]\n",
            "Generating val split: 1583 examples [00:00, 78500.63 examples/s]\n",
            "Generating test split: 3166 examples [00:00, 56907.38 examples/s]\n",
            "Map: 100% 11426/11426 [00:00<00:00, 211999.21 examples/s]\n",
            "Map: 100% 1583/1583 [00:00<00:00, 192479.58 examples/s]\n",
            "Map: 100% 3166/3166 [00:00<00:00, 242068.77 examples/s]\n",
            "INFO:root:Loaded processed sentiment Dataset\n",
            "INFO:root:Created data collator\n",
            "/usr/local/lib/python3.10/dist-packages/torch/nn/modules/rnn.py:82: UserWarning: dropout option adds dropout after all but last recurrent layer, so non-zero dropout expects num_layers greater than 1, but got dropout=0.2 and num_layers=1\n",
            "  warnings.warn(\"dropout option adds dropout after all but last \"\n",
            "INFO:root:Initialized model for training\n",
            "INFO:root:Training started...\n",
            "frist time training\n",
            "/usr/local/lib/python3.10/dist-packages/torch/utils/data/dataloader.py:557: UserWarning: This DataLoader will create 4 worker processes in total. Our suggested max number of worker in current system is 2, which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.\n",
            "  warnings.warn(_create_warning_msg(\n",
            "{'loss': 0.7665, 'learning_rate': 0.0009999606161386313, 'epoch': 1.0}\n",
            "{'eval_loss': 0.604600727558136, 'eval_accuracy': 0.7511054958938724, 'eval_f1': 0.5079549823767479, 'eval_runtime': 0.3767, 'eval_samples_per_second': 4201.981, 'eval_steps_per_second': 18.581, 'epoch': 1.0}\n",
            "/usr/local/lib/python3.10/dist-packages/torch/utils/data/dataloader.py:557: UserWarning: This DataLoader will create 4 worker processes in total. Our suggested max number of worker in current system is 2, which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.\n",
            "  warnings.warn(_create_warning_msg(\n",
            "{'loss': 0.4959, 'learning_rate': 0.0009999212322772625, 'epoch': 2.0}\n",
            "{'eval_loss': 0.45968401432037354, 'eval_accuracy': 0.8477574226152874, 'eval_f1': 0.5782110293470932, 'eval_runtime': 0.3709, 'eval_samples_per_second': 4268.289, 'eval_steps_per_second': 18.874, 'epoch': 2.0}\n",
            "/usr/local/lib/python3.10/dist-packages/torch/utils/data/dataloader.py:557: UserWarning: This DataLoader will create 4 worker processes in total. Our suggested max number of worker in current system is 2, which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.\n",
            "  warnings.warn(_create_warning_msg(\n",
            "{'loss': 0.4187, 'learning_rate': 0.0009998818484158935, 'epoch': 3.0}\n",
            "{'eval_loss': 0.42578575015068054, 'eval_accuracy': 0.8566013897662665, 'eval_f1': 0.5843101020693687, 'eval_runtime': 0.6135, 'eval_samples_per_second': 2580.457, 'eval_steps_per_second': 11.411, 'epoch': 3.0}\n",
            "/usr/local/lib/python3.10/dist-packages/torch/utils/data/dataloader.py:557: UserWarning: This DataLoader will create 4 worker processes in total. Our suggested max number of worker in current system is 2, which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.\n",
            "  warnings.warn(_create_warning_msg(\n",
            "{'loss': 0.3797, 'learning_rate': 0.0009998424645545248, 'epoch': 4.0}\n",
            "{'eval_loss': 0.40114840865135193, 'eval_accuracy': 0.8591282375236892, 'eval_f1': 0.5856689726717798, 'eval_runtime': 0.364, 'eval_samples_per_second': 4348.627, 'eval_steps_per_second': 19.23, 'epoch': 4.0}\n",
            "/usr/local/lib/python3.10/dist-packages/torch/utils/data/dataloader.py:557: UserWarning: This DataLoader will create 4 worker processes in total. Our suggested max number of worker in current system is 2, which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.\n",
            "  warnings.warn(_create_warning_msg(\n",
            "{'loss': 0.3495, 'learning_rate': 0.000999803080693156, 'epoch': 5.0}\n",
            "{'eval_loss': 0.38104304671287537, 'eval_accuracy': 0.8730259001895135, 'eval_f1': 0.5954051238033129, 'eval_runtime': 0.3792, 'eval_samples_per_second': 4174.66, 'eval_steps_per_second': 18.46, 'epoch': 5.0}\n",
            "/usr/local/lib/python3.10/dist-packages/torch/utils/data/dataloader.py:557: UserWarning: This DataLoader will create 4 worker processes in total. Our suggested max number of worker in current system is 2, which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.\n",
            "  warnings.warn(_create_warning_msg(\n",
            "{'loss': 0.3281, 'learning_rate': 0.0009997636968317872, 'epoch': 6.0}\n",
            "{'eval_loss': 0.3670244514942169, 'eval_accuracy': 0.8742893240682249, 'eval_f1': 0.5962713239385034, 'eval_runtime': 0.3647, 'eval_samples_per_second': 4340.752, 'eval_steps_per_second': 19.195, 'epoch': 6.0}\n",
            "/usr/local/lib/python3.10/dist-packages/torch/utils/data/dataloader.py:557: UserWarning: This DataLoader will create 4 worker processes in total. Our suggested max number of worker in current system is 2, which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.\n",
            "  warnings.warn(_create_warning_msg(\n",
            "{'loss': 0.3111, 'learning_rate': 0.0009997243129704183, 'epoch': 7.0}\n",
            "{'eval_loss': 0.37437981367111206, 'eval_accuracy': 0.8635502210991788, 'eval_f1': 0.5884372410106727, 'eval_runtime': 0.3576, 'eval_samples_per_second': 4426.849, 'eval_steps_per_second': 19.575, 'epoch': 7.0}\n",
            "/usr/local/lib/python3.10/dist-packages/torch/utils/data/dataloader.py:557: UserWarning: This DataLoader will create 4 worker processes in total. Our suggested max number of worker in current system is 2, which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.\n",
            "  warnings.warn(_create_warning_msg(\n",
            "{'loss': 0.2962, 'learning_rate': 0.0009996849291090495, 'epoch': 8.0}\n",
            "{'eval_loss': 0.3581385612487793, 'eval_accuracy': 0.8755527479469362, 'eval_f1': 0.6142171123390461, 'eval_runtime': 0.368, 'eval_samples_per_second': 4301.084, 'eval_steps_per_second': 19.019, 'epoch': 8.0}\n",
            "/usr/local/lib/python3.10/dist-packages/torch/utils/data/dataloader.py:557: UserWarning: This DataLoader will create 4 worker processes in total. Our suggested max number of worker in current system is 2, which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.\n",
            "  warnings.warn(_create_warning_msg(\n",
            "{'loss': 0.2777, 'learning_rate': 0.0009996455452476807, 'epoch': 9.0}\n",
            "{'eval_loss': 0.3541993796825409, 'eval_accuracy': 0.8711307643714467, 'eval_f1': 0.6267935077463597, 'eval_runtime': 0.5945, 'eval_samples_per_second': 2662.675, 'eval_steps_per_second': 11.774, 'epoch': 9.0}\n",
            "/usr/local/lib/python3.10/dist-packages/torch/utils/data/dataloader.py:557: UserWarning: This DataLoader will create 4 worker processes in total. Our suggested max number of worker in current system is 2, which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.\n",
            "  warnings.warn(_create_warning_msg(\n",
            "{'loss': 0.2659, 'learning_rate': 0.000999606161386312, 'epoch': 10.0}\n",
            "{'eval_loss': 0.3455914556980133, 'eval_accuracy': 0.8761844598862919, 'eval_f1': 0.6686063692188479, 'eval_runtime': 0.3761, 'eval_samples_per_second': 4209.451, 'eval_steps_per_second': 18.614, 'epoch': 10.0}\n",
            "/usr/local/lib/python3.10/dist-packages/torch/utils/data/dataloader.py:557: UserWarning: This DataLoader will create 4 worker processes in total. Our suggested max number of worker in current system is 2, which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.\n",
            "  warnings.warn(_create_warning_msg(\n",
            "{'loss': 0.2571, 'learning_rate': 0.0009995667775249432, 'epoch': 11.0}\n",
            "{'eval_loss': 0.3428206443786621, 'eval_accuracy': 0.883133291219204, 'eval_f1': 0.690571380262102, 'eval_runtime': 0.3667, 'eval_samples_per_second': 4316.564, 'eval_steps_per_second': 19.088, 'epoch': 11.0}\n",
            "/usr/local/lib/python3.10/dist-packages/torch/utils/data/dataloader.py:557: UserWarning: This DataLoader will create 4 worker processes in total. Our suggested max number of worker in current system is 2, which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.\n",
            "  warnings.warn(_create_warning_msg(\n",
            "{'loss': 0.2466, 'learning_rate': 0.0009995273936635744, 'epoch': 12.0}\n",
            "{'eval_loss': 0.33865150809288025, 'eval_accuracy': 0.886923562855338, 'eval_f1': 0.7052852433841502, 'eval_runtime': 0.3703, 'eval_samples_per_second': 4274.967, 'eval_steps_per_second': 18.904, 'epoch': 12.0}\n",
            "/usr/local/lib/python3.10/dist-packages/torch/utils/data/dataloader.py:557: UserWarning: This DataLoader will create 4 worker processes in total. Our suggested max number of worker in current system is 2, which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.\n",
            "  warnings.warn(_create_warning_msg(\n",
            "{'loss': 0.238, 'learning_rate': 0.0009994880098022055, 'epoch': 13.0}\n",
            "{'eval_loss': 0.3369958698749542, 'eval_accuracy': 0.886923562855338, 'eval_f1': 0.7057344436747917, 'eval_runtime': 0.5243, 'eval_samples_per_second': 3019.413, 'eval_steps_per_second': 13.352, 'epoch': 13.0}\n",
            "/usr/local/lib/python3.10/dist-packages/torch/utils/data/dataloader.py:557: UserWarning: This DataLoader will create 4 worker processes in total. Our suggested max number of worker in current system is 2, which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.\n",
            "  warnings.warn(_create_warning_msg(\n",
            "{'loss': 0.232, 'learning_rate': 0.0009994486259408367, 'epoch': 14.0}\n",
            "{'eval_loss': 0.3392639756202698, 'eval_accuracy': 0.8843967150979154, 'eval_f1': 0.6915903371274544, 'eval_runtime': 0.3645, 'eval_samples_per_second': 4342.631, 'eval_steps_per_second': 19.203, 'epoch': 14.0}\n",
            "/usr/local/lib/python3.10/dist-packages/torch/utils/data/dataloader.py:557: UserWarning: This DataLoader will create 4 worker processes in total. Our suggested max number of worker in current system is 2, which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.\n",
            "  warnings.warn(_create_warning_msg(\n",
            "{'loss': 0.2258, 'learning_rate': 0.000999409242079468, 'epoch': 15.0}\n",
            "{'eval_loss': 0.3373529016971588, 'eval_accuracy': 0.8875552747946936, 'eval_f1': 0.7062609677540039, 'eval_runtime': 0.5938, 'eval_samples_per_second': 2665.873, 'eval_steps_per_second': 11.788, 'epoch': 15.0}\n",
            "/usr/local/lib/python3.10/dist-packages/torch/utils/data/dataloader.py:557: UserWarning: This DataLoader will create 4 worker processes in total. Our suggested max number of worker in current system is 2, which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.\n",
            "  warnings.warn(_create_warning_msg(\n",
            "{'loss': 0.2173, 'learning_rate': 0.0009993698582180992, 'epoch': 16.0}\n",
            "{'eval_loss': 0.33794113993644714, 'eval_accuracy': 0.8881869867340493, 'eval_f1': 0.7197572287885287, 'eval_runtime': 0.3753, 'eval_samples_per_second': 4217.419, 'eval_steps_per_second': 18.649, 'epoch': 16.0}\n",
            "/usr/local/lib/python3.10/dist-packages/torch/utils/data/dataloader.py:557: UserWarning: This DataLoader will create 4 worker processes in total. Our suggested max number of worker in current system is 2, which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.\n",
            "  warnings.warn(_create_warning_msg(\n",
            "{'loss': 0.2154, 'learning_rate': 0.0009993304743567302, 'epoch': 17.0}\n",
            "{'eval_loss': 0.3542066514492035, 'eval_accuracy': 0.8723941882501579, 'eval_f1': 0.7037106598456505, 'eval_runtime': 0.3797, 'eval_samples_per_second': 4168.568, 'eval_steps_per_second': 18.433, 'epoch': 17.0}\n",
            "/usr/local/lib/python3.10/dist-packages/torch/utils/data/dataloader.py:557: UserWarning: This DataLoader will create 4 worker processes in total. Our suggested max number of worker in current system is 2, which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.\n",
            "  warnings.warn(_create_warning_msg(\n",
            "{'loss': 0.2107, 'learning_rate': 0.0009992910904953614, 'epoch': 18.0}\n",
            "{'eval_loss': 0.3402862250804901, 'eval_accuracy': 0.886923562855338, 'eval_f1': 0.7189217926060031, 'eval_runtime': 0.3905, 'eval_samples_per_second': 4053.729, 'eval_steps_per_second': 17.926, 'epoch': 18.0}\n",
            "/usr/local/lib/python3.10/dist-packages/torch/utils/data/dataloader.py:557: UserWarning: This DataLoader will create 4 worker processes in total. Our suggested max number of worker in current system is 2, which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.\n",
            "  warnings.warn(_create_warning_msg(\n",
            "{'loss': 0.2072, 'learning_rate': 0.0009992517066339927, 'epoch': 19.0}\n",
            "{'eval_loss': 0.33736297488212585, 'eval_accuracy': 0.8919772583701832, 'eval_f1': 0.7234267058772396, 'eval_runtime': 0.3773, 'eval_samples_per_second': 4195.348, 'eval_steps_per_second': 18.552, 'epoch': 19.0}\n",
            "/usr/local/lib/python3.10/dist-packages/torch/utils/data/dataloader.py:557: UserWarning: This DataLoader will create 4 worker processes in total. Our suggested max number of worker in current system is 2, which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.\n",
            "  warnings.warn(_create_warning_msg(\n",
            "{'loss': 0.1997, 'learning_rate': 0.000999212322772624, 'epoch': 20.0}\n",
            "{'eval_loss': 0.34265437722206116, 'eval_accuracy': 0.8907138344914719, 'eval_f1': 0.7179984732724631, 'eval_runtime': 0.3789, 'eval_samples_per_second': 4177.804, 'eval_steps_per_second': 18.474, 'epoch': 20.0}\n",
            "/usr/local/lib/python3.10/dist-packages/torch/utils/data/dataloader.py:557: UserWarning: This DataLoader will create 4 worker processes in total. Our suggested max number of worker in current system is 2, which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.\n",
            "  warnings.warn(_create_warning_msg(\n",
            "{'loss': 0.196, 'learning_rate': 0.000999172938911255, 'epoch': 21.0}\n",
            "{'eval_loss': 0.33710816502571106, 'eval_accuracy': 0.8888186986734049, 'eval_f1': 0.7103082507403101, 'eval_runtime': 0.5446, 'eval_samples_per_second': 2906.639, 'eval_steps_per_second': 12.853, 'epoch': 21.0}\n",
            "/usr/local/lib/python3.10/dist-packages/torch/utils/data/dataloader.py:557: UserWarning: This DataLoader will create 4 worker processes in total. Our suggested max number of worker in current system is 2, which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.\n",
            "  warnings.warn(_create_warning_msg(\n",
            "{'loss': 0.1915, 'learning_rate': 0.0009991335550498862, 'epoch': 22.0}\n",
            "{'eval_loss': 0.34114623069763184, 'eval_accuracy': 0.886923562855338, 'eval_f1': 0.7198367113563204, 'eval_runtime': 0.3628, 'eval_samples_per_second': 4362.782, 'eval_steps_per_second': 19.292, 'epoch': 22.0}\n",
            "/usr/local/lib/python3.10/dist-packages/torch/utils/data/dataloader.py:557: UserWarning: This DataLoader will create 4 worker processes in total. Our suggested max number of worker in current system is 2, which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.\n",
            "  warnings.warn(_create_warning_msg(\n",
            "{'loss': 0.1896, 'learning_rate': 0.0009990941711885174, 'epoch': 23.0}\n",
            "{'eval_loss': 0.3438638150691986, 'eval_accuracy': 0.8919772583701832, 'eval_f1': 0.7235335185153385, 'eval_runtime': 0.3694, 'eval_samples_per_second': 4285.748, 'eval_steps_per_second': 18.952, 'epoch': 23.0}\n",
            "/usr/local/lib/python3.10/dist-packages/torch/utils/data/dataloader.py:557: UserWarning: This DataLoader will create 4 worker processes in total. Our suggested max number of worker in current system is 2, which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.\n",
            "  warnings.warn(_create_warning_msg(\n",
            "{'loss': 0.1853, 'learning_rate': 0.0009990547873271487, 'epoch': 24.0}\n",
            "{'eval_loss': 0.3445492684841156, 'eval_accuracy': 0.8913455464308275, 'eval_f1': 0.711260258864557, 'eval_runtime': 0.3562, 'eval_samples_per_second': 4444.395, 'eval_steps_per_second': 19.653, 'epoch': 24.0}\n",
            "/usr/local/lib/python3.10/dist-packages/torch/utils/data/dataloader.py:557: UserWarning: This DataLoader will create 4 worker processes in total. Our suggested max number of worker in current system is 2, which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.\n",
            "  warnings.warn(_create_warning_msg(\n",
            "{'loss': 0.1834, 'learning_rate': 0.00099901540346578, 'epoch': 25.0}\n",
            "{'eval_loss': 0.3421753942966461, 'eval_accuracy': 0.8888186986734049, 'eval_f1': 0.7113379186616821, 'eval_runtime': 0.4255, 'eval_samples_per_second': 3720.396, 'eval_steps_per_second': 16.452, 'epoch': 25.0}\n",
            "/usr/local/lib/python3.10/dist-packages/torch/utils/data/dataloader.py:557: UserWarning: This DataLoader will create 4 worker processes in total. Our suggested max number of worker in current system is 2, which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.\n",
            "  warnings.warn(_create_warning_msg(\n",
            "{'loss': 0.1795, 'learning_rate': 0.000998976019604411, 'epoch': 26.0}\n",
            "{'eval_loss': 0.34364959597587585, 'eval_accuracy': 0.8951358180669615, 'eval_f1': 0.7131246340197311, 'eval_runtime': 0.3659, 'eval_samples_per_second': 4326.899, 'eval_steps_per_second': 19.133, 'epoch': 26.0}\n",
            "/usr/local/lib/python3.10/dist-packages/torch/utils/data/dataloader.py:557: UserWarning: This DataLoader will create 4 worker processes in total. Our suggested max number of worker in current system is 2, which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.\n",
            "  warnings.warn(_create_warning_msg(\n",
            "{'loss': 0.1762, 'learning_rate': 0.0009989366357430422, 'epoch': 27.0}\n",
            "{'eval_loss': 0.34425464272499084, 'eval_accuracy': 0.8900821225521163, 'eval_f1': 0.7146499790216986, 'eval_runtime': 0.5625, 'eval_samples_per_second': 2814.17, 'eval_steps_per_second': 12.444, 'epoch': 27.0}\n",
            "/usr/local/lib/python3.10/dist-packages/torch/utils/data/dataloader.py:557: UserWarning: This DataLoader will create 4 worker processes in total. Our suggested max number of worker in current system is 2, which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.\n",
            "  warnings.warn(_create_warning_msg(\n",
            "{'loss': 0.1725, 'learning_rate': 0.0009988972518816734, 'epoch': 28.0}\n",
            "{'eval_loss': 0.3476717472076416, 'eval_accuracy': 0.8881869867340493, 'eval_f1': 0.7311677445446696, 'eval_runtime': 0.3663, 'eval_samples_per_second': 4321.618, 'eval_steps_per_second': 19.11, 'epoch': 28.0}\n",
            "/usr/local/lib/python3.10/dist-packages/torch/utils/data/dataloader.py:557: UserWarning: This DataLoader will create 4 worker processes in total. Our suggested max number of worker in current system is 2, which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.\n",
            "  warnings.warn(_create_warning_msg(\n",
            "{'loss': 0.1712, 'learning_rate': 0.0009988578680203046, 'epoch': 29.0}\n",
            "{'eval_loss': 0.3468157947063446, 'eval_accuracy': 0.8818698673404928, 'eval_f1': 0.7232040454592847, 'eval_runtime': 0.3709, 'eval_samples_per_second': 4268.294, 'eval_steps_per_second': 18.874, 'epoch': 29.0}\n",
            "/usr/local/lib/python3.10/dist-packages/torch/utils/data/dataloader.py:557: UserWarning: This DataLoader will create 4 worker processes in total. Our suggested max number of worker in current system is 2, which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.\n",
            "  warnings.warn(_create_warning_msg(\n",
            "{'loss': 0.1667, 'learning_rate': 0.0009988184841589359, 'epoch': 30.0}\n",
            "{'eval_loss': 0.34770718216896057, 'eval_accuracy': 0.8888186986734049, 'eval_f1': 0.7203154557923895, 'eval_runtime': 0.3733, 'eval_samples_per_second': 4240.258, 'eval_steps_per_second': 18.75, 'epoch': 30.0}\n",
            "/usr/local/lib/python3.10/dist-packages/torch/utils/data/dataloader.py:557: UserWarning: This DataLoader will create 4 worker processes in total. Our suggested max number of worker in current system is 2, which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.\n",
            "  warnings.warn(_create_warning_msg(\n",
            "{'loss': 0.1654, 'learning_rate': 0.0009987791002975671, 'epoch': 31.0}\n",
            "{'eval_loss': 0.35367441177368164, 'eval_accuracy': 0.8907138344914719, 'eval_f1': 0.7346061558303426, 'eval_runtime': 0.3925, 'eval_samples_per_second': 4033.445, 'eval_steps_per_second': 17.836, 'epoch': 31.0}\n",
            "/usr/local/lib/python3.10/dist-packages/torch/utils/data/dataloader.py:557: UserWarning: This DataLoader will create 4 worker processes in total. Our suggested max number of worker in current system is 2, which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.\n",
            "  warnings.warn(_create_warning_msg(\n",
            "{'loss': 0.1619, 'learning_rate': 0.0009987397164361981, 'epoch': 32.0}\n",
            "{'eval_loss': 0.350362628698349, 'eval_accuracy': 0.8894504106127605, 'eval_f1': 0.7191664992297904, 'eval_runtime': 0.3772, 'eval_samples_per_second': 4197.178, 'eval_steps_per_second': 18.56, 'epoch': 32.0}\n",
            "/usr/local/lib/python3.10/dist-packages/torch/utils/data/dataloader.py:557: UserWarning: This DataLoader will create 4 worker processes in total. Our suggested max number of worker in current system is 2, which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.\n",
            "  warnings.warn(_create_warning_msg(\n",
            "{'loss': 0.1598, 'learning_rate': 0.0009987003325748294, 'epoch': 33.0}\n",
            "{'eval_loss': 0.35286247730255127, 'eval_accuracy': 0.883133291219204, 'eval_f1': 0.714738910382753, 'eval_runtime': 0.5905, 'eval_samples_per_second': 2680.835, 'eval_steps_per_second': 11.855, 'epoch': 33.0}\n",
            "/usr/local/lib/python3.10/dist-packages/torch/utils/data/dataloader.py:557: UserWarning: This DataLoader will create 4 worker processes in total. Our suggested max number of worker in current system is 2, which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.\n",
            "  warnings.warn(_create_warning_msg(\n",
            "{'loss': 0.1591, 'learning_rate': 0.0009986609487134606, 'epoch': 34.0}\n",
            "{'eval_loss': 0.35389137268066406, 'eval_accuracy': 0.8856601389766267, 'eval_f1': 0.7269827611183542, 'eval_runtime': 0.3887, 'eval_samples_per_second': 4072.837, 'eval_steps_per_second': 18.01, 'epoch': 34.0}\n",
            "/usr/local/lib/python3.10/dist-packages/torch/utils/data/dataloader.py:557: UserWarning: This DataLoader will create 4 worker processes in total. Our suggested max number of worker in current system is 2, which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.\n",
            "  warnings.warn(_create_warning_msg(\n",
            "{'loss': 0.1543, 'learning_rate': 0.0009986215648520918, 'epoch': 35.0}\n",
            "{'eval_loss': 0.35320401191711426, 'eval_accuracy': 0.8913455464308275, 'eval_f1': 0.7263707616900184, 'eval_runtime': 0.38, 'eval_samples_per_second': 4165.511, 'eval_steps_per_second': 18.42, 'epoch': 35.0}\n",
            "/usr/local/lib/python3.10/dist-packages/torch/utils/data/dataloader.py:557: UserWarning: This DataLoader will create 4 worker processes in total. Our suggested max number of worker in current system is 2, which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.\n",
            "  warnings.warn(_create_warning_msg(\n",
            "{'loss': 0.1535, 'learning_rate': 0.0009985821809907229, 'epoch': 36.0}\n",
            "{'eval_loss': 0.35674723982810974, 'eval_accuracy': 0.8894504106127605, 'eval_f1': 0.72079046200779, 'eval_runtime': 0.369, 'eval_samples_per_second': 4289.544, 'eval_steps_per_second': 18.968, 'epoch': 36.0}\n",
            "{'train_runtime': 85.1623, 'train_samples_per_second': 13416.74, 'train_steps_per_second': 52.84, 'train_loss': 0.24736434824672746, 'epoch': 36.0}\n",
            "/usr/local/lib/python3.10/dist-packages/torch/utils/data/dataloader.py:557: UserWarning: This DataLoader will create 4 worker processes in total. Our suggested max number of worker in current system is 2, which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.\n",
            "  warnings.warn(_create_warning_msg(\n",
            "{'eval_loss': 0.35367441177368164, 'eval_accuracy': 0.8907138344914719, 'eval_f1': 0.7346061558303426, 'eval_runtime': 0.4047, 'eval_samples_per_second': 3911.557, 'eval_steps_per_second': 17.297, 'epoch': 36.0}\n",
            "INFO:root:Training complete\n",
            "INFO:root:Metrics saved\n",
            "INFO:root:predicting...\n",
            "/usr/local/lib/python3.10/dist-packages/torch/nn/modules/rnn.py:82: UserWarning: dropout option adds dropout after all but last recurrent layer, so non-zero dropout expects num_layers greater than 1, but got dropout=0.2 and num_layers=1\n",
            "  warnings.warn(\"dropout option adds dropout after all but last \"\n",
            "INFO:root:Loading the sentiment_analysis model...\n",
            "INFO:root:best checkpoint at path: checkpoint/lstm/checkpoint-1395/pytorch_model.bin\n",
            "INFO:root:Obtaining predictions...\n",
            "accuracy on test: 0.8780795957043588\n",
            "f1 macro on test: 0.7124322082324531\n",
            "f1 weighted on test: 0.8706920625159538\n",
            "confusion matrix:\n",
            " [[1311   21   77]\n",
            " [  72   42   53]\n",
            " [ 144   19 1427]]\n",
            "INFO:root:task done!!!\n"
          ]
        }
      ],
      "source": [
        "%cd /content/\n",
        "!python ./DS201/TH5/main.py --config ./DS201/TH5/config/config.yaml"
      ]
    }
  ],
  "metadata": {
    "accelerator": "GPU",
    "colab": {
      "collapsed_sections": [
        "KvAhbK_UC4Lx"
      ],
      "provenance": []
    },
    "kernelspec": {
      "display_name": "Python 3",
      "name": "python3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 0
}
